src/app/ui-components/selectFeeRate/index.tsx

```diff
@@ -0,0 +1,176 @@
+  align-items: center;
+`;
+
+const Label = styled.span<{
+  $size: 'm' | 's';
+  $variant: 'action' | 'light' | 'mid' | 'dark';
+  $clickable?: boolean;
+}>`
+  ${(props) =>
+    props.$size === 'm'
+      ? props.theme.typography.body_medium_m
+      : props.theme.typography.body_medium_s};
+  color: ${(props) =>
+    props.$variant === 'action'
+      ? props.theme.colors.tangerine
+      : props.$variant === 'light'
+      ? props.theme.colors.white_0
+      : props.$variant === 'mid'
+      ? props.theme.colors.white_200
+      : props.theme.colors.white_400};
+  margin-bottom: ${(props) => props.theme.spacing(2)}px;
+  cursor: ${(props) => (props.$clickable ? 'pointer' : 'default')};
+
+  user-select: ${(props) => (props.$clickable ? 'none' : 'auto')};
+
+  ${(props) =>
+    props.$clickable &&
+    `
+  &:hover {
+    color: ${props.theme.colors.tangerine_200};
+  }`}
+`;
+
+type Props = {
+  fee: string | undefined;
+  feeUnits: string;
+  feeRate: string;
+  setFeeRate: (feeRate: string) => void;
+  feeRateUnits: string;
+  fiatUnit: string;
+  baseToFiat: (base: string) => string;
+  getFeeForFeeRate: (feeRate: number) => Promise<number | undefined>;
+  isLoading?: boolean;
+  feeRates: {
+    low?: number;
+    medium?: number;
+    high?: number;
+  };
+  feeRateLimits?: {
+    min?: number;
+    max?: number;
+  };
+};
+
+function SelectFeeRate({
+  fee,
+  feeUnits,
+  feeRate,
+  feeRateUnits,
+  fiatUnit,
+  baseToFiat,
+  setFeeRate,
+  getFeeForFeeRate,
+  feeRates,
+  feeRateLimits,
+  isLoading,
+}: Props) {
+  const { t } = useTranslation('translation');
+  const [editing, setEditing] = useState(false);
+
+  const feeRateSpeed = useMemo(() => {
+    if (feeRate) {
+      if (feeRate === feeRates.high?.toString()) {
+        return t('TRANSACTION_SETTING.PRIORITIES.HIGH');
+      }
+      if (feeRate === feeRates.medium?.toString()) {
+        return t('TRANSACTION_SETTING.PRIORITIES.MEDIUM');
+      }
+      if (feeRate === feeRates.low?.toString()) {
+        return t('TRANSACTION_SETTING.PRIORITIES.LOW');
+      }
+    }
+
+    return t('TRANSACTION_SETTING.PRIORITIES.CUSTOM');
+  }, [feeRates, feeRate]);
+
+  return (
+    <div>
+      <RowContainer>
+        <Label $size="m" $variant="mid">
+          {t('TRANSACTION_SETTING.NETWORK_FEE')}
+        </Label>
+        <NumericFormat
+          value={fee || '-'}
+          displayType="text"
+          thousandSeparator
+          renderText={(value: string) => (
+            <Label $size="m" $variant="light">
+              {value} {feeUnits}
+            </Label>
+          )}
+        />
+      </RowContainer>
+      <RowContainer>
+        <div>
+          <Label $size="m" $variant="dark">
+            {feeRateSpeed}{' '}
+          </Label>
+          <Label
+            $size="m"
+            $variant="action"
+            $clickable={!isLoading}
+            onClick={() => setEditing((cur) => !cur)}
+          >
+            {t('COMMON.EDIT')} <Pencil />
```

Comment:
should be size=16 and vertically centered

Reply:
fixed like this: 
![image](https://github.com/secretkeylabs/xverse-web-extension/assets/6109710/db83359e-3c41-4c2f-9fd7-537b218a4d57)